tests/unit/httpd_test.cc

```diff
@@ -1997,3 +1997,73 @@ SEASTAR_THREAD_TEST_CASE(test_http_with_broken_wire) {
 
     c.close().get();
 }
+
+SEASTAR_TEST_CASE(test_client_close_connection) {
+    return async([] {
+        condition_variable cv;
+        loopback_connection_factory lcf(1);
+        static constexpr size_t _128KiB = 128 * 1024;
+        auto cln = http::experimental::client(std::make_unique<loopback_http_factory>(lcf), 1, http::experimental::client::retry_requests::yes);
+        auto make_test_request = [&cv, &cln]() {
+            size_t content_length = 0;
+            auto req = http::request::make("GET", "test", "/test");
+            cln.make_request(
+                   std::move(req),
+                   [&cv, &content_length](const http::reply& resp, input_stream<char>&& in) {
+                       content_length = resp.content_length;
+                       return seastar::async([&cv, &content_length, in = std::move(in)]() mutable {
+                           auto buff = in.read().get();
```

Comment:
Worth having a check that the amt of bytes read is less than the claimed content length